p2p/host/basic/basic_host.go

```diff
@@ -41,13 +43,16 @@ type BasicHost struct {
 	natmgr  *natManager
 
 	proc goprocess.Process
+
+	bwc metrics.Reporter
 }
 
 // New constructs and sets up a new *BasicHost with given Network
-func New(net inet.Network, opts ...Option) *BasicHost {
+func New(net inet.Network, bwc metrics.Reporter, opts ...Option) *BasicHost {
```

Comment:
Actually, i think this is probably easier to do by aggregating outside-- instead of passing one reporter in, let these things account their own bandwidth and query it outside:

```
type Host interface {
  Bandwidth() BandwidthStats
}
```